src/rgw/driver/rados/topic_migration.h

```diff
@@ -0,0 +1,37 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright contributors to the Ceph project
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#pragma once
+
+#include <boost/asio/io_context.hpp>
+#include <spawn/spawn.hpp>
+
+class DoutPrefixProvider;
+namespace rgw::sal { class RadosStore; }
+
+// the squid release changes the format of topic/notification metadata. once the
+// notification_v2 feature gets enabled, this migration logic runs on startup to
+// convert all v1 metadata to the v2 format
+namespace rgwrados::topic_migration {
+
+int migrate(const DoutPrefixProvider* dpp,
+            rgw::sal::RadosStore* driver,
+            boost::asio::io_context& context,
+            spawn::yield_context yield);
+
+int deconstruct_topics_oid(const std::string& bucket_topics_oid, std::string& tenant, std::string& bucket_name,
+                                  std::string& marker, const DoutPrefixProvider* dpp);
```

Comment:
this doesn't need to be exposed by the header. just move its definition inside of the "anonymous namespace" block in topic_migration.cc

Reply:
Ack!